clash-lib/src/Clash/Driver/Types.hs

```diff
@@ -110,6 +110,9 @@ data ClashOpts = ClashOpts { opt_inlineLimit :: Int
                            -- ^ Enable aggressive X optimization, which may
                            -- remove undefineds from generated HDL by replaced
                            -- with defined alternatives.
+                           , opt_inlineWFCacheLimit :: Word
+                           -- ^ When do we start caching transformed work-free
+                           -- top-level binders.
```

Comment:
I like the other version better :-)

```-- ^ At what size do we cache normalized? work-free top-level binders```

Maybe add "Determined by heuristic `termSize`"